class.ext_update.php

```diff
@@ -280,4 +290,135 @@ protected function updateMetadataConfig() {
 
     }
 
+    /**
+     * Check all configured Solr cores
+     *
+     * @access	protected
+     *
+     * @return	boolean
+     */
+    protected function solariumSolrUpdateRequired() {
+
+        // Get all Solr cores that were not deleted.
+        $result = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
+            'index_name',
+            'tx_dlf_solrcores',
+            'deleted = 0',
+            '',
+            '',
+            ''
+            );
+
+        while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
+
+            // Instantiate search object.
+            $solr = tx_dlf_solr::getInstance($resArray['index_name']);
+
+            if (!$solr->ready) {
+
+                return TRUE;
+
+            }
+
+        }
+
+        return FALSE;
+
+    }
+
+    /**
+     * Create all configured Solr cores
+     *
+     * @access	protected
+     *
+     * @return	void
+     */
+    protected function doSolariumSolrUpdate() {
+
+        // Get all Solr cores that were not deleted.
+        $result = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
+            'index_name',
+            'tx_dlf_solrcores',
+            'deleted = 0',
+            '',
+            '',
+            ''
+            );
+
+        while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
+
+            // Instantiate search object.
+            $solr = tx_dlf_solr::getInstance($resArray['index_name']);
+
+            if (!$solr->ready) {
+
+                $conf = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['dlf']);
+
+                $solrInfo = tx_dlf_solr::getSolrConnectionInfo();
+
+                // Prepend username and password to hostname.
+                if ($solrInfo['username'] && $solrInfo['password']) {
+
+                    $host = $solrInfo['username'].':'.$solrInfo['password'].'@'.$solrInfo['host'];
+
+                } else {
+
+                    $host = $solrInfo['host'];
+
+                }
+
+                $context = stream_context_create(array (
+                    'http' => array (
+                        'method' => 'GET',
+                        'user_agent' => ($conf['useragent'] ? $conf['useragent'] : ini_get('user_agent'))
+                    )
+                ));
+
+                // Build request for adding new Solr core.
+                // @see http://wiki.apache.org/solr/CoreAdmin
+                $url = $solrInfo['scheme'].'://'.$host.':'.$solrInfo['port'].'/'.$solrInfo['path'].'/admin/cores?wt=xml&action=CREATE&name=dlfCore'.$resArray['index_name'].'&instanceDir=dlfCore'.$resArray['index_name'].'&dataDir=data&configSet=dlf';
```

Comment:
The resulting cores are named "dlfCoredlfCore0". It must be "dlfCore0" instead.

Reply:
Fixed